### ISSUE TITLE
Incorrect handling geo-queries

### ISSUE THREAD

**Issue by Antiavanti** [florianholzapfel/express-restify-mongoose#187]
Query from client:
```javascript
/api/v1/collection?query={
	"geometry": {
		"$geoIntersects": {
			"$geometry": {
				"type": "Polygon",
				"coordinates": [[
					[16.14560058593753,51.307365718453234],
					[18.02151611328128,51.307365718453234],
					[18.02151611328128,50.79110589223593],
					[16.14560058593753,50.79110589223593],
					[16.14560058593753,51.307365718453234]
				]]
			}
		}
	}
}
```
Inside of req._ermQueryOptions:
```javascript
{
	"query": {
		"geometry": {
			"$geoIntersects": {
				"$geometry": {
					"type": "Polygon",
					"coordinates": {
						"$in": [{
							"$in": [{
								"$in": [16.092042236328155,51.35713184352065]
							},
							{
								"$in": [17.967957763671905,51.35713184352065]
							},
							{
								"$in": [17.967957763671905,50.841430047612135]
							},
							{
								"$in": [16.092042236328155,50.841430047612135]
							},
							{
								"$in": [16.092042236328155,51.35713184352065]
							}]
						}]
					}
				}
			}
		}
	}
}
```
Those $in operators are wrong :)

**Comment by Zertz** [comment]
Interesting, if you want to take a stab at it, this is happening [here](https://github.com/florianholzapfel/express-restify-mongoose/blob/master/lib/middleware/prepareQuery.js#L27).

**Comment by Antiavanti** [comment]
Currently I'm handling it in pre middleware (change it back only for geojsons).
But adding $in operator in this particular case is just incorrect behavior :)

**Comment by Zertz** [comment]
Yes I was just pointing out the faulty code, feel free to send a PR :)

**Comment by muliyul** [comment]
This is confirmed to happen in 4.0.2.

My query: 
![image](https://cloud.githubusercontent.com/assets/8338082/14970403/430b55ba-10d1-11e6-8f5d-334deb46fed6.png)

The result:
![image](https://cloud.githubusercontent.com/assets/8338082/14970368/0067ea98-10d1-11e6-8c91-14f72dc19e84.png)

**Comment by Zertz** [comment]
Is the result of `sw.lng()` and `sw.lat()` a number?

**Comment by muliyul** [comment]
I posted the result I'm getting in devtools... so yeah I'm pretty sure. also I double checked.

**Comment by Zertz** [comment]
Sorry, I somehow completely overlooked the second screenshot. The issue is that we look specifically for numbers, which should be fixed in [is-coordinates](https://github.com/Zertz/is-coordinates/blob/master/index.js#L9).

**Comment by muliyul** [comment]
Do you have any release date for this?

**Comment by Zertz** [comment]
Probably sometimes this week, but nothing is set in stone.

**Comment by muliyul** [comment]
Any news on this issue @Zertz ?

**Comment by Zertz** [comment]
I've been quite busy lately, though I will take some open-source time Friday and/or this weekend :-)

**Comment by muliyul** [comment]
I'm counting on you! I've tried to fix it but I couldn't figure out how. By the way I'm not entirely sure this has anything to do with strings as I am sending my data as number and it is verified by the is-coordinates. I'm sending a query with the `type` param to specify a polygon type query and it looks like the query is being understood by ERM as something entirely different.

**Comment by muliyul** [comment]
I feel really bad to nudge about this, but I may have to use another library. Deadline incoming... thank you for trying @Zertz

**Comment by MaxVinogradov** [comment]
@Zertz have you any success with this issue? I'm going to use this library, and seems will deal with coordinates.
